packages/nextcloud/lib/src/helpers/spreed.dart

```diff
@@ -0,0 +1,145 @@
+// ignore_for_file: public_member_api_docs
+import 'package:nextcloud/src/api/core.openapi.dart' as core;
+import 'package:nextcloud/src/api/spreed.openapi.dart' as spreed;
+import 'package:version/version.dart';
+
+/// Version of the spreed app supported
+const supportedVersion = 17;
+
+extension SpreedVersionSupported on spreed.Client {
+  /// Check if the spreed app version is supported by this client
+  ///
+  /// Also returns the supported version number
+  Future<(bool, int)> isSupported(final core.OcsGetCapabilitiesResponseApplicationJson_Ocs_Data capabilities) async {
+    final version = capabilities.capabilities.spreedPublicCapabilities?.spreed.version;
+    return (
+      version != null && Version.parse(version).major == supportedVersion,
+      supportedVersion,
+    );
+  }
+}
+
+enum RoomType {
+  @Deprecated('Do not use')
+  stub,
+  oneToOne,
+  group,
+  public,
+  changelog;
+
+  bool get isSingleUser => this == RoomType.oneToOne || this == RoomType.changelog;
+}
+
+enum MessageType {
+  comment,
+  // ignore: constant_identifier_names
+  comment_deleted,
+  system,
+  command;
+}
+
+enum ActorType {
+  users,
+  guests,
+  bots,
+  bridged;
+}
+
+enum ParticipantType {
+  @Deprecated('Do not use')
+  stub,
+  owner,
+  moderator,
+  user,
+  guest,
+  userFollowingPublicLink,
+  guestWithModeratorPermissions,
+}
+
+enum ParticipantPermission {
+  $default(0),
+  custom(1),
+  startCall(2),
+  joinCall(4),
+  canIgnoreLobby(8),
+  canPublishAudio(16),
+  canPublishVideo(32),
+  canScreenShare(64),
+  canSendMessageAndShareAndReact(128);
+
+  const ParticipantPermission(this.code);
+  final int code;
```

Comment:
Please again use bit operations here.
Just make it a getter that shifts `1` by the index of the element.

```dart
int get value => 1 << index; 
```